pkg/operator/ceph/cluster/mon/service.go

```diff
@@ -67,6 +69,20 @@ func (c *Cluster) createService(mon *monConfig) (string, error) {
 		return "", nil
 	}
 
-	logger.Infof("mon %q ip is %s", mon.DaemonName, s.Spec.ClusterIP)
+	if c.spec.Network.MultiClusterService.Enabled {
+		exportedIP, err := k8sutil.ExportService(c.ClusterInfo.Context, c.context, s)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to export service %q", s.Name)
+		}
+		logger.Infof("mon %q exported IP is %s", mon.DaemonName, exportedIP)
+
+		// remove mon canary deployment only after the service is exported because DNS
+		// query on <service>.<ns>.svc.clusterset.local requires the mon pod be running
```

Comment:
We need to remove the canary pod individually else the mon pod will fail to start due to pod-anti-affinity rules.